posts/2022-09-27-instant-on-220011-beta.adoc

```diff
@@ -0,0 +1,1171 @@
+----
+
+Note the amount of time it took for Liberty to report it has been started and check out the service running in the container at the http://localhost:9080/dev/system/properties URL. After you are finished checking out the application, stop the running container by pressing CTRL+C in the command-line session where executed the `podman run` command.
+
+[#checkpoint-app]
+=== Checkpoint an application in-container
+
+Open Liberty has three defined places (phases) during the startup process where a checkpoint can occur:
+
+1. `features` - This is the earliest phase during startup where a checkpoint can happen.  The checkpoint will occur after all of the configured Open Liberty features have been started.  This is before any processing occurs for the installed applications.
+2. `deployment` - The checkpoint will happen after processing the configured applications meta-data.  If the application has any components that get run as part of the application starting then the checkpoint will be taken before executing any code from the application.
+3. `applications` - This is the last phase where a checkpoint can happen, implying that this phase has the potential for providing the fastest startup time when restoring the application instance.  The checkpoint will happen after all configured applications have been reported as started.  This phase happens before opening any ports for listening to incoming requests for the applications.
+
+The `applications` phase typically provides the quickest startup time for an application, but it also may cause some application code to run before the server process checkpoint happens.  This may lead to undesired behavior when restoring the checkpoint process if the application holds on to some state that should not be restored into more than one concurrent instance of the application.  For example, connecting to an outside resource such as a database before the checkpoint is taken will result in a failure to restore many instances of such a process since this would try to restore the same connection multiple times. If your application initialization does not perform such operations, e.g. 
+open database connections, it may be possible to use the applications checkpoint in that case.
+
+Once an application container image has been built it can be used to checkpoint the application process at one of the specified checkpoint phases described above (`features`, `deployment`, `applications`). This is done by using the `--env` option to `podman run` to set the value for `WLP_CHECKPOINT` to one of the available checkpoint phases. For this example, use the `applications` phase by running the following `podman` command:
+
+.Perform a checkpoint in container
+[source]
+----
+podman run \
+  --name getting-started-checkpoint-container \
+  --privileged \
+  --env WLP_CHECKPOINT=applications \
+  getting-started
+----
+- The `--privileged` option is required to perform the `criu` checkpoint in-container.
```

Comment:
Because it is the easiest option to get up and running.  This is just running a container locally so there really isn't that much concern.  It is when in production that you really don't want a privileged container.